### PR TITLE
README + tutorial

### PR DESCRIPTION
**Pull request by paraynaud**

### CODE REVIEW

**Comment by paraynaud** [comment]
Show issue of `ExpressionTreeTools`, must be fix with [v0.1.1](https://github.com/JuliaSmoothOptimizers/ExpressionTreeForge.jl/pull/28)

**Comment by paraynaud** [comment]
@dpo I think the tutorial is ready to be read.

**Comment by paraynaud** [comment]
@dpo I did apply your remarks, tell me if it is okay for you now.

**Comment by dpo** [comment]
If I click on "tutorial" on the 1st page, I get a page not found.

File: README.md [hunk 1]
```diff
@@ -17,74 +17,21 @@
...
-This structure allows to define a partitioned quasi-Newton approximation of $\nabla^2 f$
-$$
-B = \sum_{i=1}^N U_i^\top \hat{B}_{i} U_i
-$$
-such that each $\hat{B}_i \approx \nabla^2 \hat{f}_i$.
-Contrary to the BFGS and SR1 updates, a partitioned quasi-Newton approximation keeps the sparsity structure of $\nabla^2 f$, moreover, the rank of update $B$ is proportionnal to $\min(N,n)$ while unstructured quasi-Newton update's rank is 1 or 2.
-#### Reference
-* A. Griewank and P. Toint, [*Partitioned variable metric updates for large structured optimization problems*](10.1007/BF01399316), Numerische Mathematik volume, 39, pp. 119--137, 1982.
+## Philosophy
+PartiallySeparableSolvers.jl define partitioned quasi-Newton trust-region solvers for unconstrained `ADNLPModel` or `MathOptNLPModel`.
```
Comment (dpo) [inline README.md:22]:
```suggestion
PartiallySeparableSolvers.jl implements a partitioned quasi-Newton trust-region solvers for unconstrained optimization.
```

File: README.md [hunk 2]
```diff
@@ -17,74 +17,21 @@
...
 
-## How to install
-```
-julia> ]
-pkg> add https://github.com/paraynaud/PartitionedStructures.jl, https://github.com/paraynaud/ExpressionTreeForge.jl, https://github.com/paraynaud/PartiallySeparableNLPModels.jl, https://github.com/paraynaud/PartiallySeparableSolvers.jl
-pkg> test PartiallySeparableSolvers
-```
\ No newline at end of file
+## Dependencies
+The module uses [PartiallySeparableNLPModels.jl](https://github.com/JuliaSmoothOptimizers/PartiallySeparableNLPModels.jl) to detect the partially-separable structure and the structure required to partitioned quasi-Newton approximations.
```
Comment (dpo) [inline README.md:37]:
```suggestion
The module uses [PartiallySeparableNLPModels.jl](https://github.com/JuliaSmoothOptimizers/PartiallySeparableNLPModels.jl) to detect the partially-separable structure and the structure required for partitioned quasi-Newton approximations.
```

File: docs/src/index.md [hunk 3]
```diff
@@ -1 +1,19 @@
-# PartiallySeparableSolvers.jl
\ No newline at end of file
+# PartiallySeparableSolvers.jl
+
+## Philosophy
+PartiallySeparableSolvers.jl define partitioned quasi-Newton trust-region solvers for unconstrained `ADNLPModel` or `MathOptNLPModel`.
```
Comment (dpo) [inline docs/src/index.md:4]:
```suggestion
PartiallySeparableSolvers.jl implements a partitioned quasi-Newton trust-region solver for unconstrained optimization.
```

File: docs/src/tutorial.md [hunk 4]
```diff
@@ -1 +1,76 @@
...
+```math
+\nabla^2 f(x) = \sum_{i=1}^N U_i^\top \nabla^2 f_i (U_i x) U_i,
+```
+to maintain a partitioned a quasi-Newton approximation
+```math
+B = \sum_{i=1}^N U_i^\top B_{i} U_i,
+```
+where each $B_{i} \approx \nabla^2 f_i$ has size $n_i \times n_i$.
+
+The usual quasi-Newton updates, such as SR1, BFGS, or their limited-memory variants LSR1 or LBFGS approximate $\nabla^2 f(x)\approx B$ by low rank updates producing inevitably dense approximations.
```
Comment (dpo) [inline docs/src/tutorial.md:24]:
```suggestion
The usual quasi-Newton updates, such as SR1, BFGS, or their limited-memory variants LSR1 or LBFGS, approximate $\nabla^2 f(x)\approx B$ by low rank updates, which inevitably result in dense approximations.
```

File: docs/src/tutorial.md [hunk 5]
```diff
@@ -1 +1,76 @@
...
+By relying on element Hessian approximations $B_i$, updated at each iteration, the partitioned updates respect the Hessian sparsity structure and perform updates of rank $\min(\lambda N,n), \; \lambda = 1,2$ depending on the update applied to $B_i$.
```
Comment (dpo) [inline docs/src/tutorial.md:25]:
```suggestion
By relying on element Hessian approximations $B_i$ updated at each iteration, the partitioned updates respect the Hessian sparsity structure and perform updates of rank $\min(\lambda N,n), \; \lambda = 1,2$ depending on the update applied to $B_i$.
```

File: docs/src/tutorial.md [hunk 6]
```diff
@@ -1 +1,76 @@
...
+
+See [PartitionedStructures.jl tutorial](https://JuliaSmoothOptimizers.github.io/PartitionedStructures.jl/dev/tutorial/) to get more details about partitioned quasi-Newton approximations and how they compare with standard updates.
```
Comment (dpo) [inline docs/src/tutorial.md:27]:
```suggestion
See [PartitionedStructures.jl tutorial](https://JuliaSmoothOptimizers.github.io/PartitionedStructures.jl/dev/tutorial/) for more details about partitioned quasi-Newton approximations and how they compare with standard updates.
```

File: docs/src/tutorial.md [hunk 7]
```diff
@@ -1 +1,76 @@
...
+Some of these partitioned quasi-Newton methods are detailed in the reference below, and some are new (see PartitionedStructures.jl).
+#### Reference
```
Comment (dpo) [inline docs/src/tutorial.md:29]:
```suggestion

#### Reference
```

File: docs/src/tutorial.md [hunk 8]
```diff
@@ -1 +1,76 @@
...
+* A. Griewank and P. Toint, [*Partitioned variable metric updates for large structured optimization problems*](https://link.springer.com/article/10.1007/BF01399316), Numerische Mathematik volume, 39, pp. 119--137, 1982.
```
Comment (dpo) [inline docs/src/tutorial.md:30]:
```suggestion
* A. Griewank and Ph. L. Toint, [*Partitioned variable metric updates for large structured optimization problems*](https://link.springer.com/article/10.1007/BF01399316), Numerische Mathematik volume, 39, pp. 119--137, 1982.
```

File: docs/src/tutorial.md [hunk 9]
```diff
@@ -1 +1,76 @@
...
+
+
+## Running a partitioned quasi-Newton solver
+For now, PartiallySeparableSolvers.jl supports [ADNLPModel](https://github.com/JuliaSmoothOptimizers/ADNLPModels.jl)s, which are defined by pure julia code, or [MathOptNLPModel](https://github.com/JuliaSmoothOptimizers/NLPModelsJuMP.jl)s based on [JuMP](https://github.com/jump-dev/JuMP.jl) models.
```
Comment (dpo) [inline docs/src/tutorial.md:34]:
```suggestion
For now, PartiallySeparableSolvers.jl supports [ADNLPModel](https://github.com/JuliaSmoothOptimizers/ADNLPModels.jl)s, which are defined by pure julia code, or [MathOptNLPModel](https://github.com/JuliaSmoothOptimizers/NLPModelsJuMP.jl)s, which are based on [JuMP](https://github.com/jump-dev/JuMP.jl) models.
```

File: docs/src/tutorial.md [hunk 10]
```diff
@@ -1 +1,76 @@
...
+function example(x)
+  n = length(x)
+  n < 2 && @error("length of x must be >= 2")
+  return sum((x[j] + x[j+1])^2 for j=1:n-1)
+end 
+start_example(n :: Int) = ones(n)
+example_ADNLPModel(n :: Int=100) = ADNLPModel(example, start_example(n), name="Example " * string(n) * " variables")
+
+n = 10 # size of the problem
+adnlp_model = example_ADNLPModel(n)
```
Comment (dpo) [inline docs/src/tutorial.md:51]:
```suggestion
model = ADNLPModel(f, ones(10))
```

File: docs/src/tutorial.md [hunk 11]
```diff
@@ -1 +1,76 @@
...
+## Running a partitioned quasi-Newton solver
+For now, PartiallySeparableSolvers.jl supports [ADNLPModel](https://github.com/JuliaSmoothOptimizers/ADNLPModels.jl)s, which are defined by pure julia code, or [MathOptNLPModel](https://github.com/JuliaSmoothOptimizers/NLPModelsJuMP.jl)s based on [JuMP](https://github.com/jump-dev/JuMP.jl) models.
+Regardless of which model is used, the solver returns a [`GenericExecutionStats`](https://juliasmoothoptimizers.github.io/SolverCore.jl/dev/reference/#SolverCore.GenericExecutionStats).
+
+### An `ADNLPModel`
+Let's first define the function that we seek to minimize and wrap it into an `ADNLPModel`:
+```@example PSSolver
+using ADNLPModels
+
+function example(x)
```
Comment (dpo) [inline docs/src/tutorial.md:42]:
```suggestion
function f(x)
```

File: docs/src/tutorial.md [hunk 12]
```diff
@@ -1 +1,76 @@
...
+
+n = 10 # size of the problem
+adnlp_model = example_ADNLPModel(n)
+```
+
+### Running the partitioned quasi-Newton trust-region method
+You minimize your model by calling the partitioned-update solver `PTRUNK`:
+```@example PSSolver
+using PartiallySeparableSolvers
+stats = PTRUNK(adnlp_model)
```
Comment (dpo) [inline docs/src/tutorial.md:58]:
```suggestion
stats = PTRUNK(model)
```

Comment (dpo) [reply docs/src/tutorial.md:58]: Montre la sortie du solveur (propre) svp.

File: docs/src/tutorial.md [hunk 13]
```diff
@@ -1 +1,76 @@
...
+The partitioned quasi-Newton update performed by `PTRUNK` is chosen by the optional argument `name`.
+Allowed values include:
+- `name=:pbfgs` each $B_i$ is updated by BFGS;
+- `name=:psr1` each $B_i$ is updated by SR1;
+- `name=:pse` each $B_i$ may be updated by both BFGS and SR1;
+- `name=:plbfgs` each $B_i$ is an LBFGS operator;
+- `name=:plsr1` each $B_i$ is an LSR1 operator;
+- `name=:plse` (by default) each $B_i$ may be an LBFGS or LSR1 operator.
+
+```@example PSSolver
```
Comment (dpo) [inline docs/src/tutorial.md:71]:
```suggestion
Thus for example,
```@example PSSolver
```

File: docs/src/tutorial.md [hunk 14]
```diff
@@ -1 +1,76 @@
...
+stats = PTRUNK(mathopt_model; name=:pbfgs)
+print(stats)
+```
```
Comment (dpo) [inline docs/src/tutorial.md:74]:
```suggestion
```
updates each element Hessian approximation with BFGS.
```

**Review by dpo — CHANGES_REQUESTED** [review]

File: docs/src/tutorial.md (hunk 12 above)
Comment (dpo) [reply docs/src/tutorial.md]: Non, le détail des itérations (utilise Trunk comme modèle).

**Pull request merged**